### PR TITLE
Upgrade to at least boost-1.74.0

### LINKED ISSUE #1512: Raise required Boost version to 1.74
In today's meeting we have decided to raise the required Boost version to 1.74 (as requested in https://github.com/alpaka-group/alpaka/pull/1453#issuecomment-987665863).

### PR DESCRIPTION
**Pull request by bernhardmgruber**
Fixes #1512

### CODE REVIEW

**Comment by psychocoderHPC** [comment]
Can we please finalize this change after #1520. Because both will collide.

**Comment by bernhardmgruber** [comment]
It turns out clang-5 does not support Boost.Fibre: https://github.com/boostorg/fiber/issues/272. I disabled the backend for clang-5.

**Comment by j-stephan** [comment]
This should also be caught in our CMake scripts, then.

**Comment by bernhardmgruber** [comment]
> This should also be caught in our CMake scripts, then.

I did that as well.

**Comment by bernhardmgruber** [comment]
In the CMake scripts or in the CI scripts? AFAIK we don't check for such stuff in the CMake scripts. I added a check to the CI scripts.

**Comment by j-stephan** [comment]
Huh? We do this all the time:

https://github.com/alpaka-group/alpaka/blob/7064fcd05d9ad149bbbdc04b9aa96c7ad9302461/cmake/alpakaCommon.cmake#L291-L292

https://github.com/alpaka-group/alpaka/blob/7064fcd05d9ad149bbbdc04b9aa96c7ad9302461/cmake/alpakaCommon.cmake#L296-L298

https://github.com/alpaka-group/alpaka/blob/7064fcd05d9ad149bbbdc04b9aa96c7ad9302461/cmake/alpakaCommon.cmake#L344-L346

etc.

**Comment by bernhardmgruber** [comment]
I see. Thx for sharing! I remembered we had checks on which CUDA versions where compatible with which compilers and I couldn't find those anymore in alpakaCommon.cmake. So I figured we don't have such checks anymore.

**Comment by j-stephan** [comment]
Rebase please!

File: script/before_install.sh [hunk 1]
```diff
@@ -149,20 +149,17 @@ then
             exit 1
         fi
     fi
+fi
 
-    if [ "${ALPAKA_CI_STDLIB}" == "libstdc++" ]
+if [ ! -z "${ALPAKA_CI_CLANG_VER+x}" ]
+then
+    if [ "${ALPAKA_CI_CLANG_VER}" == 5 ]
```
Comment (psychocoderHPC) [inline script/before_install.sh:164]: Do we still support clang-5? I think this check can be removed full.

**Review by psychocoderHPC — CHANGES_REQUESTED** [review]

File: script/before_install.sh (hunk 1 above)
Comment (bernhardmgruber) [reply script/before_install.sh:164]: Unfortunately, we still have clang-5.

Comment (psychocoderHPC) [reply script/before_install.sh:164]: Ok no problem, I thought clang-5 is not supporting C++17.

**Review by psychocoderHPC — DISMISSED** [review]

**Comment by psychocoderHPC** [comment]
The CI was already green but produced now a merge conflict with the changes in the dev branch. I will merge this pr without passing the CI. CI will check this branch in the develop branch again.

**Review by psychocoderHPC — APPROVED** [review]

**Pull request merged**